src/Microsoft.AspNetCore.Mvc.Razor/Internal/RazorPagePropertyActivator.cs

```diff
@@ -30,19 +30,21 @@ public class RazorPagePropertyActivator
         {
             _metadataProvider = metadataProvider;
 
+            // In the absence of a model on the current type, we'll attempt to use ViewDataDictionary<object> on the current type.
+            var viewDataDictionaryModelType = modelType ?? typeof(object);
```

Comment:
For the following scenario:

```
// /_ViewImports.cshtml
@inherits CustomBasePage

// _ViewStart.cshtml
@"Hello world"

// Page.cshtml
@page
@model TestModel
```

The model value ends up being `null` for the `_ViewStart` since no `Model` property is available on it. However we're starting with a `VDD<TestModel>`. This results in an exception when we attempt to initialize the page. The fix here is that if you're attempting to initialize a page without a model property, you will get a `VDD<object>`